### PR TITLE
[MAINTENANCE] remove integration markers

### PR DESCRIPTION
**Pull request by Kilo59**
Remove all `pytest` `integration` markers.
Add `docs` to the marker matrix.

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -85,6 +85,7 @@ jobs:
           # `not e2e` because 1 test & 1 fixture need access token
           - cloud and not e2e
           - filesystem
+          - docs
```
Comment (Kilo59) [inline .github/workflows/ci.yml:88]: adding `docs`

File: pyproject.toml [hunk 2]
```diff
@@ -541,7 +541,6 @@ markers = [
     "e2e: mark test as an E2E test.",
     "external_sqldialect: mark test as requiring install of an external sql dialect.",
     "filesystem: mark tests using the filesystem as the storage backend.",
-    "integration: mark test as an integration test.",
```
Comment (Kilo59) [inline pyproject.toml:544]: removing the `integration` marker from the registered pytest markers section

File: ci/azure-pipelines-docs-integration.yml [hunk 3]
```diff
@@ -112,7 +112,7 @@ stages:
           displayName: 'Install dependencies'
 
         - script: |
-            pytest -v --docs-tests -m integration --spark tests/integration/test_script_runner.py
```
Comment (Kilo59) [inline ci/azure-pipelines-docs-integration.yml:115]: Running these without `-m integration`

File: tasks.py [hunk 4]
```diff
@@ -797,11 +797,18 @@ class TestDependencies(NamedTuple):
     "cloud": TestDependencies(
         ("reqs/requirements-dev-cloud.txt",), exta_pytest_args=("--cloud",)
     ),
+    "docs": TestDependencies(
+        requirement_files=(
+            "reqs/requirements-dev-test.txt",
+            "reqs/requirements-dev-spark.txt",
+        ),
+        exta_pytest_args=("docs-tests",),
+    ),
```
Comment (Kilo59) [inline tasks.py:806]: Had to add this `docs` mapping so that the `--docs-tests` flag would be used.

File: tasks.py [hunk 5]
```diff
@@ -797,11 +797,18 @@ class TestDependencies(NamedTuple):
     "cloud": TestDependencies(
         ("reqs/requirements-dev-cloud.txt",), exta_pytest_args=("--cloud",)
     ),
+    "docs": TestDependencies(
+        requirement_files=(
+            "reqs/requirements-dev-test.txt",
+            "reqs/requirements-dev-spark.txt",
+        ),
+        exta_pytest_args=("--docs-tests",),
+    ),
```
Comment (Kilo59) [inline tasks.py:806]: Had to add this `docs` mapping so that the `--docs-tests` flag would be used.

**Review by Shinnnyshinshin — APPROVED** [review]
LGTM

**Comment by Shinnnyshinshin** [comment]
Hi @Kilo59 it looks like the `static-analysis` is failing. Could you have a look?

**Comment by Kilo59** [comment]
@Shinnnyshinshin that lint failure is from @billdirks base branch.

**Review by billdirks — APPROVED** [review]
Thank you!

**Pull request merged**